src/Common/src/System/Dynamic/Utils/DelegateHelpers.cs

```diff
@@ -59,8 +62,15 @@ private static Delegate CreateObjectArrayDelegateRefEmit(Type delegateType, Syst
             LocalBuilder retValue = ilgen.DeclareLocal(typeof(object));
 
             // create the argument array
-            ilgen.Emit(OpCodes.Ldc_I4, parameters.Length);
-            ilgen.Emit(OpCodes.Newarr, typeof(object));
+            if (parameters.Length == 0)
+            {
+                ilgen.Emit(OpCodes.Call, s_ArrayEmpty);
```

Comment:
This is only ever used for in-memory assemblies / execution, right?  I'm just confirming, because otherwise we'd need to make sure that we only used Array.Empty when running against a framework that had it.